common/authentication/postgresql/metadata.go

```diff
@@ -43,36 +43,38 @@ func (m *PostgresAuthMetadata) Reset() {
 	m.ConnectionMaxIdleTime = 0
 	m.MaxConns = 0
 	m.UseAzureAD = false
+	m.UseAWSIAM = false
 	m.QueryExecMode = ""
 }
 
 // InitWithMetadata inits the object with metadata from the user.
 // Set azureADEnabled to true if the component can support authentication with Azure AD.
 // This is different from the "useAzureAD" property from the user, which is provided by the user and instructs the component to authenticate using Azure AD.
-func (m *PostgresAuthMetadata) InitWithMetadata(meta map[string]string, azureADEnabled bool) (err error) {
+func (m *PostgresAuthMetadata) InitWithMetadata(meta map[string]string, azureADEnabled, awsIAMEnabled bool) (err error) {
 	// Validate input
-	if m.ConnectionString == "" {
-		return errors.New("missing connection string")
-	}
-
-	// Populate the Azure environment if using Azure AD
-	if azureADEnabled && m.UseAzureAD {
+	switch {
+	case azureADEnabled && m.UseAzureAD:
 		m.azureEnv, err = azure.NewEnvironmentSettings(meta)
 		if err != nil {
 			return err
 		}
-	} else {
-		// Make sure this is false
+	case awsIAMEnabled && m.UseAWSIAM:
+		// Do y'all want the logic moved to moreso here instead of where I have it?
+	case m.ConnectionString == "":
+		return errors.New("missing connection string")
```

Comment:
At least for Azure, and I suspect AWS too, the connection string must not be empty even when using Azure AD